src/main/java/mate/academy/spring/boot/repository/book/spec/AuthorSpecificationProvider.java

```diff
@@ -0,0 +1,19 @@
+package mate.academy.spring.boot.repository.book.spec;
+
+import java.util.Arrays;
+import mate.academy.spring.boot.model.Book;
+import mate.academy.spring.boot.repository.SpecificationProvider;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AuthorSpecificationProvider implements SpecificationProvider<Book> {
```

Comment:
Could we join different specification providers in one BookSpecificationProvider and implement `public Specification<Book> titleContainsIgnoreCase(List<String> titles)` and `public Specification<Book> authorIn(List<String> authors)` and isbn there ?

Reply:
With AuthorSpecificationProvider everythings good. I just suggest to join providers into one BookSpecificationProvider and implement methods there: title, author, isdn.

Reply:
I still don't understand... We search according to one or more parameters, in my case it is three fields, and not according to the entire entity. I tried to do it like this: `package mate.academy.spring.boot.repository.book.spec;

import lombok.RequiredArgsConstructor;
import mate.academy.spring.boot.model.Book;
import mate.academy.spring.boot.repository.SpecificationProvider;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.stereotype.Component;

@RequiredArgsConstructor
@Component
public class BookSpecificationProvider implements SpecificationProvider<Book> {

    @Override
    public String getKey() {
        return null;
    }

    @Override
    public Specification<Book> getSpecification(String[] params) {
        return null;
    }

    @Component
    private static class TitleSpecification implements SpecificationProvider<Book> {
        private static final String TITLE_KEY = "title";

        @Override
        public String getKey() {
            return TITLE_KEY;
        }

        @Override
        public Specification<Book> getSpecification(String[] params) {
            return (root, query, criteriaBuilder) -> root.get("title").in((Object[]) params);
        }
    }

    private static class AuthorSpecification implements SpecificationProvider<Book> {
        private static final String AUTHOR_KEY = "author";

        @Override
        public String getKey() {
            return AUTHOR_KEY;
        }

        @Override
        public Specification<Book> getSpecification(String[] params) {
            return (root, query, criteriaBuilder) -> root.get("author").in((Object[]) params);
        }
    }

    @Component
    private static class IsbnSpecification implements SpecificationProvider<Book> {
        private static final String ISBN_KEY = "isbn";

        @Override
        public String getKey() {
            return ISBN_KEY;
        }

        @Override
        public Specification<Book> getSpecification(String[] params) {
            return (root, query, criteriaBuilder) -> root.get("isbn").in((Object[]) params);
        }
    }
}
` But then the search works like the getAll method  for the author and title, and ISBN generally gives a 500 error
And when you try to return everything to individual implementations, the result is somehow the same again ... (the search works like the PUEFdd method for the author and title, and ISBN generally gives a 500 error)
in the end I was completely confused